### ISSUE TITLE
No such file or directory verbs.txt

### ISSUE THREAD

**Issue by slayer** [jeromedalbert/rubocop-obsession#6]
Hey

```
bundle exec rubocop -D -E --format=html -o tmp/rubocop.html --format=progress app/ lib/
No such file or directory @ rb_sysopen - /usr/local/bundle/ruby/3.2.0/gems/rubocop-obsession-0.1.5/lib/rubocop/cop/mixin/files/verbs.txt
/usr/local/bundle/ruby/3.2.0/gems/rubocop-obsession-0.1.5/lib/rubocop/cop/mixin/helpers.rb:6:in `read'
/usr/local/bundle/ruby/3.2.0/gems/rubocop-obsession-0.1.5/lib/rubocop/cop/mixin/helpers.rb:6:in `<module:Helpers>'
/usr/local/bundle/ruby/3.2.0/gems/rubocop-obsession-0.1.5/lib/rubocop/cop/mixin/helpers.rb:5:in `<module:Cop>'
```

**Comment by jeromedalbert** [comment]
@slayer I cannot repro, would you mind running `bundle env` and sharing the output?

**Comment by slayer** [comment]
Hey

this issue occurs on our CI (gitlab)

here is `bundle env`


```
$ bundle env
## Environment
Bundler       2.4.10
  Platforms   ruby, x86_64-linux
Ruby          3.2.6p234 (2024-10-30 revision 63aeb018eb1cc0f7b00f955980711fd1bd94af7f) [x86_64-linux]
  Full Path   /usr/local/bin/ruby
  Config Dir  /usr/local/etc
RubyGems      3.4.19
  Gem Home    /usr/local/bundle
  Gem Path    /home/REDACTED/.local/share/gem/ruby/3.2.0:/usr/local/lib/ruby/gems/3.2.0:/usr/local/bundle
  User Home   /home/REDACTED
  User Path   /home/REDACTED/.local/share/gem/ruby/3.2.0
  Bin Dir     /usr/local/bundle/bin
Tools         
  Git         2.30.2
  RVM         not installed
  rbenv       not installed
  chruby      not installed
## Bundler Build Metadata
Built At          2023-03-27
Git SHA           7ffdec80d0
Released Version  true
```
## Bundler settings
```
app_config
  Set via BUNDLE_APP_CONFIG: "/usr/local/bundle"
gemfile
  Set via BUNDLE_GEMFILE: "/REDACTED/Gemfile"
jobs
  Set for your local app (/usr/local/bundle/config): 16
path
  Set via BUNDLE_PATH: "/usr/local/bundle"
silence_root_warning
  Set via BUNDLE_SILENCE_ROOT_WARNING: true
system
  Set for your local app (/usr/local/bundle/config): "true"
without
  Set for your local app (/usr/local/bundle/config): [:development, :guard]
```
## Gemfile
### Gemfile
```ruby
source 'https://rubygems.org'/
gem 'rails', '~> 7'
....
```

**Comment by jeromedalbert** [comment]
Thanks, I'm guessing this is the `bundle env` info for your CI env right? This way I can try to repro closest to the CI setup.

**Comment by slayer** [comment]
right, this output from CI, yes

**Comment by jeromedalbert** [comment]
I wasn't able to repro but I made this branch https://github.com/jeromedalbert/rubocop-obsession/tree/verbs-tentative-fix with this commit https://github.com/jeromedalbert/rubocop-obsession/commit/79be20092796ba0a7c0f93a81fc102438b9f587a that lazy loads the `verbs.txt`, maybe that'll work better.

@slayer Could you:

1. Try this branch on your CI with something like this in your Gemfile:

    ```ruby
    gem 'rubocop-obsession',
        require: false,
        github: 'jeromedalbert/rubocop-obsession',
        branch: 'verbs-tentative-fix'
    ```

   And maybe it'll just work.

2. If it still doesn't work and you get the same error, continue using that branch, but also disable cops that check verbs in your `.rubocop.yml` like so:

    ```yml
    Obsession/Rails/ServiceName:
      Enabled: false
    Obsession/Graphql/MutationName:
      Enabled: false
    ```

Hopefully solution 1 just works, otherwise you could do solution 2 combined with solution 1, which wouldn't be ideal since it wouldn't fix the root cause, but at least you'd be able to run other cops like `MethodOrder` etc.

Either way let me know if either solution worked for you; if so I will merge the branch. Thanks!

---

<details>
 <summary>(Details on how I wasn't able to repro)</summary>

<br>

I spun up a Docker container with a similar setup as your `bundle env`, with a 7.2 Rails app with Ruby 3.2.6 and Bundler 2.4.10. Then I ran `bundle exec rubocop -D -E --format=html -o tmp/rubocop.html --format=progress app/ lib/`, with or without `RAILS_ENV=test` and `CI=true` (which eager loads the app in default Rails setups):

- I was not able to repro inside the container
- In the container, doing `cat /usr/local/bundle/ruby/3.2.0/gems/rubocop-obsession-0.1.5/lib/rubocop/cop/mixin/files/verbs.txt` displays the file content for me, so that file exists. Wondering if your CI somehow deletes txt files when installing gems, although that seems unlikely.

Otherwise the only thing left at this point would be for you to make a [minimal example app](https://www.codetriage.com/example_app#how-to-make-an-example-app) that would reproduce the problem, although it would be hard for you to do since it would probably work for you on your local and not in CI. And even then I might still not repro because your CI could behave in a unique way.)
</details>

**Comment by slayer** [comment]
> Wondering if your CI somehow deletes txt files when installing gems, although that seems unlikely.

it turned out that this was exactly the reason.
the piece from our Dockerfile:

```sh
  bundle install -j 16 && \
    ( cd ${BUNDLE_PATH} && \
      find . -name *.so | xargs strip && \
      find . -regex ".*\.\(gem\|h\|c\|o\|a\|md\|txt\|rdoc\|o\|pdf\)" | xargs rm && \
      find . -type d -a -regex "^.*/\(tmp\|docs\|benchmark\|spec\|samples\)$" | egrep -v "(rack/test|http/features)" | xargs rm -rf && \
      find . -type d -a -regex "^.*/\(FEATURES\|.travis.yml\)$" | xargs rm -rf )
```
sorry

**Comment by jeromedalbert** [comment]
Ok, the best fix is to change your CI, as it is not uncommon for gems to store data in txt files, for example `PublicSuffix` does it too: https://github.com/weppos/publicsuffix-ruby/blob/c84facf1d4a1dc5fff8a30447edf5242d6a6c903/data/list.txt.

On my end I released version 0.1.9 which lazy-loads the verbs file, because it looked like best practice anyways.
